planning.py

```diff
@@ -192,16 +196,43 @@ def closest_vertex(self, q, vertices=None):
 
         return v_closest, min_dist
 
+    def closest_vertex_graph(self, T):
+        """Find the closest vertex belonging to graph to a another given graph T"""
+        vertices_graph_T = T.graph
+        vertices_graph = self.graph
+        min_dist = vertices_graph[0].distance(vertices_graph_T[0])
+        v_closest = vertices_graph[0]
+        v_closest_T = vertices_graph_T[0]
+        for vertex in vertices_graph_T:
+            for v in vertices_graph:
+                d = vertex.distance(v)
+
+                if d < min_dist:
+                    min_dist = d
+                    v_closest = v
+                    v_closest_T = vertex
+        return v_closest, v_closest_T, min_dist
+
     def query(self, start, goal):
         """Get the shortest path between the start and goal points, if one exists."""
+        start_time = time.time()
         vs, _ = self.graph.closest(start)
         vg, _ = self.graph.closest(goal)
         path = self.graph.path_Astar(vs, vg)
         if path is None:
             return None
+
+        self.query_time = time.time() - start_time
         idx = path[0]
         points = np.array([self.graph[i].coord for i in idx])
-        return np.vstack((start, points, goal))
+        points = np.vstack((start, points, goal))
+        t = GraphPlanner(UGraph())
+        points = [UVertex(name=str(i), coord=i) for i in points]
+        t.graph.add_vertex(points[0])
+        for i in range(1, len(points)):
+            t.graph.add_vertex(points[i])
+            t.graph.add_edge(points[i - 1], points[i])
+        return t
```

Comment:
What was the reasoning for changing this function? As far as I can tell, the difference is that instead of an array of points, it returns a new GraphPlanner object whose graph is the path.